File: 程序员面试金典第6版/面试题 01.01. Is Unique LCCI.py
# hashmap
class Solution:
    def isUnique(self, astr: str) -> bool:
        have = {}
        for i in range(len(astr)):
            if astr[i] in have:
                return False
            have[astr[i]] = i
        return True

# bit operation
class Solution:
    def isUnique(self, astr: str) -> bool:
        mark = 0
        for c in astr:
            move_bit = ord(c)-ord('a') # a: 97 A: 65 1: 49
            if (mark & (1 << move_bit)) != 0:
                return False
            mark  |= (1 << move_bit)
        return True

File: 程序员面试金典第6版/面试题 01.01. Is Unique LCCI.cpp
class Solution {
public:
    bool isUnique(string astr) {
        for(int i=0; i<astr.length(); i++) {
            if(astr.find(astr[i]) != astr.rfind(astr[i])) {
                return false;
            }
        }
        return true;
    }
};